README.md

```diff
@@ -93,9 +93,129 @@ TODO aspect ratio + images alignment + more more images
+
+- 🐞 Shader Logic, CPU-Tested: A subset of HLSL compiles as both C++ and SPIR-V, enabling CPU-side debugging of GPU logic, ensuring correctness in complex tasks like FFT, Prefix Sum, etc.
+Future Proof: C++20 Concepts in HLSL for safe and documented Static Polymorphism
+
+- 🔮 Future-Proof: C++20 concepts in HLSL enable safe and documented polymorphism.
+
+- 🧠 Insane: Boost Preprocessor and Template Metaprogramming in HLSL!
+
+- 🛠️ Real-World Problem Solvers: The library offers GPU-optimized solutions for tasks like Prefix Sum, Binary Search, FFT, Global Sort, and even emulated `shaderFloat64` when native GPU support is unavailable!
+
+```
+[TODO][CODE] Code for each or just one showcasing most of the above points?
+```
+
+### 🧩 Full Embrace of [Buffer Device Address]() and [Descriptor Indexing]()
+
+By utilizing Buffer Device Addresses (BDAs), Nabla enables more direct access to memory through 64-bit GPU virtual addresses. Synergized with Descriptor Indexing, this approach enhances flexibility by enabling more dynamic, scalable resource binding without relying on traditional descriptor sets.
+
+### 🧩 Minimally Invasive Design
+[TODO]: vulkan handle acquisition, multiple windows, content playing second fiddle
+
+### 🧩 Designed for Interoperation
+Nabla is built with interoperation in mind, supporting memory export and import between different compute and graphics APIs.
+
+----
+
+[TODO]:
+- Cancellable Future based Async I/O
+- Virtual File System (archive mounting, our alternative to #embed, everything is referenced by absolute - path)
+
+----
+
+### 🧩 Asset Manager
+Nabla’s Asset Manager efficiently loads assets while tracking dependencies using a Directed Acyclic Graph (DAG). Assets are loaded in the correct order, avoids redundant allocations, and simplifies resource management.
+
+### 🧩 Asset Converter (CPU to GPU)
+The Asset Converter transforms CPU objects (asset::IAsset) into GPU objects (video::IBackendObject) while eliminating duplicates. Instead of relying on pointer comparisons, it hashes asset contents to detect and reuse identical GPU objects.
+
+### 🧩 Unit-Tested BxDFs for Physically Based Rendering
+A statically polymorphic library for defining Bidirectional Scattering Distribution Functions (BxDFs) in HLSL and C++. Each BxDF is rigorously unit-tested in C++ as well as HLSL. This is part of Nabla’s HLSL-C++ compatible library.
+
+### 🔧 In Progress: Property Pools (GPU Entity Component System)
+*Property Pools* group related properties together in a Structure Of Arrays (SoA) manner, allowing efficient, cache-friendly access to data on the GPU. The system enables transferring properties (Components) between the CPU and GPU, with the `PropertyPoolHandler` managing scattered updates with a special compute shader. Handles are assigned for each object and remain constant as data is added or removed.
```

Comment:
GPU ECS buzzword must be used, code examples

Reply:
Give me link to examples than we already have that cleanly portrays usage of property pools, couldn't find recent one or something that can fit in a <15loc snippet

Reply:
oops the code examples are nowhere near done :( https://github.com/Devsh-Graphics-Programming/Nabla/pull/649

so I guess we just brag about stuff thats not finished then :)